pilot/pkg/model/jwks_resolver.go

```diff
@@ -449,6 +448,7 @@ func (r *JwksResolver) refresh() bool {
 					atomic.AddUint64(&r.refreshJobFetchFailedCount, 1)
 					return
 				}
+				k.jwksURI = jwksURI
```

Comment:
you're not storing the key back to the cache.

Reply:
we are storing it here - https://github.com/istio/istio/blob/master/pilot/pkg/model/jwks_resolver.go#L462. Do you think there's a point to store it if we can't fetch the pubkey from that ?

Reply:
the problem is the old key with empty jwksURI is not removed, you're not updating it but just adding a new key, by update I mean you should replace the old one with the new content.